src/wrappers/express/ExpressWrapper.ts

```diff
@@ -0,0 +1,43 @@
+import Reporter from '../../Reporter';
+import ConfigProvider from '../../config/ConfigProvider';
+import { ApplicationManager } from '../../application/ApplicationManager';
+import ExecutionContextManager from '../../context/ExecutionContextManager';
+import * as ExpressExecutor from './ExpressExecutor';
+import WrapperUtils from '../WebWrapperUtils';
+import ExecutionContext from '../../context/ExecutionContext';
+import ThundraLogger from '../../ThundraLogger';
+
+export function expressMW(opts: any = {}) {
+    ApplicationManager.setApplicationInfoProvider().update({
+        applicationClassName: 'Express',
+        applicationDomainName: 'API',
+    });
+
+    const config = opts.config || ConfigProvider.thundraConfig;
+    const { apiKey } = config;
+    const reporter = opts.reporter || new Reporter(apiKey);
+    const pluginContext = opts.pluginContext || WrapperUtils.createPluginContext(apiKey, ExpressExecutor);
+    const plugins = opts.plugins || WrapperUtils.createPlugins(config, pluginContext);
+
+    if (!opts.disableAsyncContextManager) {
+        WrapperUtils.initAsyncContextManager();
+    }
+
+    return (req: any, res: any, next: any) => ExecutionContextManager.runWithContext(
+        WrapperUtils.createExecContext,
+        async function () {
+            const context: ExecutionContext = this;
+            try {
+                await WrapperUtils.beforeRequest(req, plugins);
```

Comment:
I think it might be better to to set a flag after `await WrapperUtils.beforeRequest(req, plugins);` so it indicates that before request completed successfully. And then `WrapperUtils.afterRequest(res, plugins, reporter);` is executed only if this flag is set. So if somehow before request fails, after request is not executed

Reply:
Ok, never mind this comment